src/System.Management.Automation/engine/remoting/common/RemoteSessionNamedPipe.cs

```diff
@@ -600,14 +606,78 @@ public void Dispose()
 
         #region Public Methods
 
+        /// <summary>
+        /// Creates the custom named pipe server with the given pipename.
+        /// </summary>
+        /// <param name="pipeName">The name of the pipe to create.</param>
+        public static void CreateCustomNamedPipeServer(string pipeName)
+        {
+            lock (s_syncObject)
+            {
+                if (_customNamedPipeServer != null && !_customNamedPipeServer.IsDisposed)
+                {
+                    if (pipeName == _customNamedPipeServer.PipeName)
+                    {
+                        // we shouldn't recreate the server object if we're using the same pipeName
+                        return;
+                    }
+
+                    // Dispose of the current pipe server so we can create a new one with the new pipeName
+                    _customNamedPipeServer.Dispose();
+                }
+
+                if (!Platform.IsWindows)
+                {
+                    int maxNameLength = (Platform.IsLinux ? _maxPipePathLengthLinux : _maxPipePathLengthMacOS) - Path.GetTempPath().Length;
+                    if (pipeName.Length > maxNameLength)
+                    {
+                        throw new InvalidOperationException(
+                            string.Format(
+                                RemotingErrorIdStrings.DebugPipeNameTooLong,
+                                maxNameLength,
+                                pipeName,
+                                pipeName.Length));
+                    }
+                }
+
+                try
+                {
+                    try
+                    {
+                        _customNamedPipeServer = new RemoteSessionNamedPipeServer(pipeName);
+                    }
+                    catch (IOException)
+                    {
+                        // Expected when named pipe server for this process already exists.
+                        // This can happen if process has multiple AppDomains hosting PowerShell (SMA.dll).
+                        return;
+                    }
+
+                    // Listener ended callback, used to create listening new pipe server.
+                    _customNamedPipeServer.ListenerEnded += OnCustomNamedPipeServerEnded;
+
+                    // Start the pipe server listening thread, and provide client connection callback.
+                    _customNamedPipeServer.StartListening(ClientConnectionCallback);
+                }
+                catch (Exception)
```

Comment:
Can we catch more specific exception?

Reply:
There's actually quite a wide range of exceptions to catch here... and since we want this to be a best attempt so that powershell can still start if something goes wrong, it's best to catch just `Exception`.